### PR TITLE
Add diagnostics for bug 1689489 (SHOW SLAVE STATUS truncates Relay_Ma…

### PR DESCRIPTION
**Pull request by laurynas-biveinis**
…ster_Log_File intermittently)

SHOW SLAVE STATUS returned
Master_Log_File	master-bin.000004
Relay_Master_Log_File	master-bin.00000

I.e. Relay_Master_Log_File is truncated by one character. Add an
assert to catch this condition.

http://jenkins.percona.com/job/percona-server-5.6-param/1871/

### CODE REVIEW

File: sql/rpl_slave.cc [hunk 1]
```diff
@@ -3168,13 +3168,23 @@ bool show_slave_status(THD* thd, Master_info* mi)
...
                     &my_charset_bin);
     protocol->store((ulonglong) mi->rli->get_group_relay_log_pos());
-    protocol->store(mi->rli->get_group_master_log_name(), &my_charset_bin);
+    const char * const relay_master_log_file=
+        mi->rli->get_group_master_log_name();
+#ifndef DBUG_OFF
+    const size_t master_log_file_len= strlen(master_log_file);
+    const size_t relay_master_log_file_len= strlen(relay_master_log_file);
+#endif
+    DBUG_ASSERT((relay_master_log_file_len == strlen(master_log_file))
```
Comment (percona-ysorokin) [inline sql/rpl_slave.cc]: (relay_master_log_file_len == master_log_file_len)?

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

**Review by robgolebiowski — APPROVED** [review]
LGTM

**Comment by laurynas-biveinis** [comment]
http://jenkins.percona.com/job/percona-server-5.6-param/1878/

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**